pty/src/serial.rs

```diff
@@ -22,71 +20,40 @@ use std::path::PathBuf;
 use std::sync::{Arc, Mutex};
 use std::time::Duration;
 
-type Handle = Arc<Mutex<SystemPort>>;
+type Handle = Arc<Mutex<SerialPort>>;
 
 pub struct SerialTty {
     port: OsString,
-    baud: BaudRate,
-    char_size: CharSize,
-    parity: Parity,
-    stop_bits: StopBits,
-    flow_control: FlowControl,
+    baud: u32,
 }
 
 impl SerialTty {
     pub fn new<T: AsRef<OsStr> + ?Sized>(port: &T) -> Self {
         Self {
             port: port.as_ref().to_owned(),
-            baud: BaudRate::Baud9600,
-            char_size: CharSize::Bits8,
-            parity: Parity::ParityNone,
-            stop_bits: StopBits::Stop1,
-            flow_control: FlowControl::FlowSoftware,
+            baud: 9600,
         }
     }
 
-    pub fn set_baud_rate(&mut self, baud: BaudRate) {
+    pub fn set_baud_rate(&mut self, baud: u32) {
         self.baud = baud;
     }
-
-    pub fn set_char_size(&mut self, char_size: CharSize) {
```

Comment:
removing these methods is a breaking API change in the pty crate, and they are important things to be able to change.

Please restore these methods and arrange for them to configure the port via https://docs.rs/serial2/latest/serial2/struct.Settings.html